framework/src/service/ServiceRegistrationBase.cpp

```diff
@@ -326,7 +332,10 @@ namespace cppmicroservices
             d->coreInfo->bundleServiceInstance.clear();
 
             d->reference = nullptr;
-            d->coreInfo->unregistering = false;
+
+            // reset d->coreInfo->unregistering to false
+            isUnregistering = true;
```

Comment:
I don't understand the logic of setting `isUnregistering` to true. At this point, isn't unregistering finished and shouldn't `isUnregistering` be `false`?

Is this solely so that `atomic_compare_exchange_strong` can be called? If so why not just say `d->coreInfo->unregistering = false;`?

Reply:
This is solely to call the atomic_compare_Exchange_strong... But I didn't like that in order to start unregistering we have to use an atomic compare, but to exit unregistration we can just set the variable... Unless that is just so that we can compare and set and setting a boolean is an atomic operation... which I dont believe it is

Reply:
`d->coreInfo->unregistering` is a `std::atomic<bool>` and the `operator=` of `atomic<bool>` is the same as [`atomic<T>::store`](https://en.cppreference.com/w/cpp/atomic/atomic/store) - https://stackoverflow.com/questions/18850752/must-i-call-atomic-load-store-explicitly

Reply:
ah ok. Yep that makes sense. Sorry, was looking at wrong variable... but yes, then I agree